### ISSUE TITLE
Interpreting output of training

### ISSUE THREAD

**Issue by Jsevillamol** [ctlearn-project/ctlearn#50]
I have run one of the benchmark configurations (concretely `config/v_0_2_0_benchmarks/LST_cnn_rnn_config.yml`) on training mode.

The beginning of the `logfile.log` produced is as follows:

```
INFO:Batch size: 16
INFO:Training and evaluating...
INFO:Total number of training events: 76860
INFO:Total number of validation events: 8541
INFO:Number of training steps per epoch: 4803
INFO:Number of training steps per validation: 2500
INFO:Total number of examples: 377098
INFO:Number of gamma (class 0) examples: 189633 (50.287%)
INFO:Number of proton (class 1) examples: 187465 (49.713%)
```

This prompts me the following questions:

1. I would have assumed that the `Total number of examples` was the summation of training and validation events, but it is not. What does it represent then? In single telescope runs it coincides with the summation of training and validation events.
2. Minor suggestion: `Number of training steps per validation` should be changed to `Number of training steps *between* validations`. I got very confused trying to understand what it meant.

After each evaluation, a line is produced saying things like `INFO:Saving dict for global step 2500: accuracy = 0.6158129, accuracy_gamma = 0.6657754, accuracy_proton = 0.5658949, auc = 0.6625105, global_step = 2500, loss = 1.3074992`.

I assume that those are the metrics on the validation set (correct?). Are they just stored in the `eval_validation/events.out.tfevents` file? 

If so, how could I produce something similar to the plots in `config/v_0_2_0_benchmark/readme.md`? The plotting scripts in `scripts` seem to expect a list of `.csv` files as an argument. I assume this would be the output of running a trained net in prediction mode, but how do I feed the validation set I used for training to the trained net for predictions?

Also, what is the easiest way of reading the training time off the output?

One more question: since the number of training steps per epoch is different but the number of epochs is the same in each `config.yml`, how come that the last checkpoint is always in step `37500`?

**Comment by aribrill** [comment]
> I would have assumed that the Total number of examples was the summation of training and validation events, but it is not. What does it represent then? In single telescope runs it coincides with the summation of traMinor suggestion: Number of training steps per validation should be changed to Number of training steps *between* validations. I got very confused trying to understand what it meant.
ining and validation events.

This is a bug, the logger output for array-level models is incorrect. See #48.

> Minor suggestion: Number of training steps per validation should be changed to Number of training steps *between* validations. I got very confused trying to understand what it meant.

Duly noted.

> How could I produce something similar to the plots in config/v_0_2_0_benchmark/readme.md? 

You can view all of the metrics and plots, and the training time, using Tensorboard, which reads from the saved output files. You can view the progress in real time or after the run is finished. See [here](https://github.com/ctlearn-project/ctlearn#run-a-model) for instructions.

> One more question: since the number of training steps per epoch is different but the number of epochs is the same in each config.yml, how come that the last checkpoint is always in step 37500?

The word "epochs" is being used in two different senses; this should be changed. In the logging output, an epoch means the number of batches/steps in the training set, while in the config file it means the number of batches/steps between validations.
